x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/LookupJoin.java

```diff
@@ -71,4 +76,36 @@ protected NodeInfo<Join> info() {
             config().rightFields()
         );
     }
+
+    @Override
+    public void postAnalysisVerification(Failures failures) {
+        super.postAnalysisVerification(failures);
+        right().forEachDown(EsRelation.class, esr -> {
+            var indexNameWithModes = esr.indexNameWithModes();
+            if (indexNameWithModes.size() != 1) {
+                failures.add(
+                    fail(
+                        esr,
+                        "invalid [{}] resolution in lookup mode to an index in [{}] indices",
```

Comment:
```suggestion
                        "invalid [{}] resolution in lookup mode to [{}] indices",
```

Previously it had a slightly different message, not sure if it was changed intentionally.